r/run_model.R

```diff
@@ -0,0 +1,25 @@
+library(optparse)
+library(readr)
+library(tidyr)
+
+# Create a CLI ------------------------------------
+# Two args: `input_dir`, `output_dir`
+option_list <- list(
+  make_option("--input_dir", type="character", default="/input", metavar="character"),
+  make_option("--output_dir", type="character", default="/output", metavar="character")
+)
+opt_parser <- OptionParser(option_list=option_list)
+opt <- parse_args(opt_parser)
+
+
+# Read in input file ------------------------------
+data <- read_csv(file.path(opt$input_dir, "names.csv"), col_types = "c")
+
+
+# Run a prediction --------------------------------
+# Full name only contains two names.
+predictions <- extract(data, name, into = c("name", "first", "last"), regex = "((.*?) (.*))")
```

Comment:
Current regex is converting 1st string to first and the rest to last name. How about `"((.*) (.*$))"`, which takes the last pattern as last name and the reset to first name.

Not a big deal here though.

Reply:
Thanks! This was on purpose, as this is how the Python script behaves. That is, it splits by whitespace, then puts the first name into "first" and the rest into "last".

Reply:
yea, I guess`John Jacob Jingleheimer Schmidt` is tricky to me lol. I usually think the last pattern (`Schmidt`) will be the last name and the rest (`John Jacob Jingleheimer`) will be the first name (+middle name). I believe you are aware of what's the right splits for the name string.

Reply:
> I usually think the last pattern (...) will be the last name and the rest (...) will be the first name (+middle name)

That's valid!  Again, this is not a sophisticated "prediction model" at all 😅 But I'm hoping this will at least give the participants some sort of idea?